lib/core/src/Ouroboros/Network/Client/Wallet.hs

```diff
@@ -595,9 +609,14 @@ localTxSubmission queue =
   where
     clientStIdle
         :: m (LocalTxClientStIdle tx err m Void)
-    clientStIdle = atomically (readTQueue queue) <&> \case
+    clientStIdle = atomically (peekTQueue queue) <&> \case
```

Comment:
It might be easier to keep using `readTQueue` but then put an `unGetTQueue` in an `onException` handler.